File: app/src/main/java/com/divercity/android/core/base/PaginatedQueryRepository.java
package com.divercity.android.core.base;

import android.support.annotation.Nullable;
import com.divercity.android.core.utils.Listing;

/**
 * Created by lucas on 01/10/2018.
 */

public interface PaginatedQueryRepository<T> {

    Listing<T> fetchData(@Nullable String query);

    void retry();

    void refresh();

    void clear();

}

File: app/src/main/java/com/divercity/android/core/base/PaginatedQueryRepository.kt
package com.divercity.android.core.base

import com.divercity.android.core.utils.Listing

/**
 * Created by lucas on 01/10/2018.
 */

interface PaginatedQueryRepository<T> {

    fun fetchData(query: String?): Listing<T>

    fun retry()

    fun refresh()

    fun clear()

}
